Add release notes for 11/7 2.14.4 (#11151)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.14.4

### New improved flow run graph with dependency layout

> The Flow Run Graph has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.

Reviewer: I don't believe this should be capitalized
```suggestion
The flow run graph in the Prefect UI has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.
```

> The Flow Run Graph has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.

Reviewer: ```suggestion
The Flow Run Graph has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts; and you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.
```

> The Flow Run Graph has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.

Reviewer: ```suggestion
The Flow Run Graph has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. We included additional small bug fixes, including the display of cached tasks.
```

> The flow run graph in the Prefect UI has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts, you can adjust it in the graph settings or with the new keyboard shortcuts - and +. Additional small bug fixes, including displaying of cached tasks, have also been included.

Reviewer: ```suggestion
The flow run graph in the Prefect UI has been rebuilt from the ground up, offering significantly improved performance capabilities that allow larger flow runs to be displayed much more smoothly. We’ve added three new layouts: two non-temporal layout options, designed to provide a clearer picture of the dependency paths, and one to facilitate easy comparison of run durations. The x-axis can now be independently scaled for temporal layouts; and you can adjust it in the graph settings or with the new keyboard shortcuts - and +. We've also included small bug fixes, including the display of cached tasks.
```
^ sounds more active

<p align="center">
<img width="976" alt="flow run graph sequential grid view" src="https://user-images.githubusercontent.com/6776415/281769376-bccc4cd5-db2c-42b9-9c21-fc32b094323b.png">
</p>

See the following pull requests for implementation details:
- Improved flow run graph with new dependency layout — https://github.com/PrefectHQ/prefect/pull/11112
- Flow run graph v2 polish — https://github.com/PrefectHQ/prefect/pull/11105
- Handle Cached tasks on graph-v2 — https://github.com/PrefectHQ/prefect/pull/11113

> - Exclude any flow/task run with a `NULL` `start_time` from Graph v2 — https://github.com/PrefectHQ/prefect/pull/11132

Reviewer: do users know about Graph v2? Can we just say "flow run graph" perhaps or something similar

> - Use an unstarted flow run's expected start time for the Graph start time — https://github.com/PrefectHQ/prefect/pull/11138

Reviewer 2: ```suggestion
- Use an unstarted flow run's expected start time as the Graph's start time — https://github.com/PrefectHQ/prefect/pull/11138
```

> - Use an unstarted flow run's expected start time as the Graph's start time — https://github.com/PrefectHQ/prefect/pull/11138

Reviewer: ```suggestion
- Use an unstarted flow run's expected start time as the flow run graph's start time — https://github.com/PrefectHQ/prefect/pull/11138
```

> See the following pull requests for implementation details:
> - Improved flow run graph with new dependency layout — https://github.com/PrefectHQ/prefect/pull/11112
> - Flow run graph v2 polish — https://github.com/PrefectHQ/prefect/pull/11105
> - Handle Cached tasks on graph-v2 — https://github.com/PrefectHQ/prefect/pull/11113
> - Exclude any flow/task run with a `NULL` `start_time` from Graph v2 — https://github.com/PrefectHQ/prefect/pull/11132
> - Use an unstarted flow run's expected start time as the Graph's start time — https://github.com/PrefectHQ/prefect/pull/11138

Reviewer: ```suggestion
See the following pull requests for implementation details:
- https://github.com/PrefectHQ/prefect/pull/11112
- https://github.com/PrefectHQ/prefect/pull/11105
- https://github.com/PrefectHQ/prefect/pull/11113
- https://github.com/PrefectHQ/prefect/pull/11132
- https://github.com/PrefectHQ/prefect/pull/11138
```

### Enhancements
- Add API route for block counts — https://github.com/PrefectHQ/prefect/pull/11090

> - Improved tag handling on `DeploymentImage` for `.deploy`

Reviewer: ```suggestion
- Improved tag handling on `DeploymentImage` for `.deploy`:
```

# Prefect Release Notes

## Release 2.14.4

### New improved flow run graph with dependency layout

We've re-written the flow run graph from scratch for maximum performance. Rendering of large graphs should be much more stable in v2 and we've added a few new features including a much requested dependency layout!

#### Features
- Grid based dependency layouts
- Comparative duration layout
- Cached tasks appear on the graph
- Faster layouts
- Optimized api endpoint for fetching graph data
- Horizontal scale controls

### Enhancements

> - Add a count blocks api route — https://github.com/PrefectHQ/prefect/pull/11090

Reviewer: ```suggestion
- Add API route for block counts — https://github.com/PrefectHQ/prefect/pull/11090
```

- Improved tag handling on DeploymentImage — https://github.com/PrefectHQ/prefect/pull/11115
- Improve tag handling in `DeploymentImage` with `parse_image_tag` — https://github.com/PrefectHQ/prefect/pull/11119

> - Allow `image` passed into `deploy` to be optional if loading flow from storage — https://github.com/PrefectHQ/prefect/pull/11117

Reviewer: ```suggestion
- Allow `image` passed into `.deploy` to be optional if loading flow from storage — https://github.com/PrefectHQ/prefect/pull/11117
```

Reviewer: this helps differentiate from `prefect deploy` cli experience

> - Ensure client avoids image builds for managed work pools — https://github.com/PrefectHQ/prefect/pull/11120

Reviewer: ```suggestion
- Ensure client avoids image builds when deploying to managed work pools — https://github.com/PrefectHQ/prefect/pull/11120
```

# Prefect Release Notes

## Release 2.14.4

### New improved flow run graph with dependency layout

We've re-written the flow run graph from scratch for maximum performance. Rendering of large graphs should be much more stable in v2 and we've added a few new features including a much requested dependency layout!

#### Features
- Grid based dependency layouts
- Comparative duration layout
- Cached tasks appear on the graph
- Faster layouts
- Optimized api endpoint for fetching graph data
- Horizontal scale controls

### Enhancements
- Add a count blocks api route — https://github.com/PrefectHQ/prefect/pull/11090

> - Improved tag handling on DeploymentImage — https://github.com/PrefectHQ/prefect/pull/11115
> - Improve tag handling in `DeploymentImage` with `parse_image_tag` — https://github.com/PrefectHQ/prefect/pull/11119

Reviewer: I think these can be combined like:

```suggestion
- Improved tag handling on `DeploymentImage` for `.deploy`
    - https://github.com/PrefectHQ/prefect/pull/11115
    - https://github.com/PrefectHQ/prefect/pull/11119
```

- Allow `image` passed into `deploy` to be optional if loading flow from storage — https://github.com/PrefectHQ/prefect/pull/11117
- Ensure client avoids image builds for managed work pools — https://github.com/PrefectHQ/prefect/pull/11120

> - Improved flow run graph with new dependency layout — https://github.com/PrefectHQ/prefect/pull/11112

Reviewer: Same with this one

- Add `SIGTERM` handling to runner to gracefully handle timeouts — https://github.com/PrefectHQ/prefect/pull/11133
- Flow run graph v2 polish — https://github.com/PrefectHQ/prefect/pull/11105
- Handle Cached tasks on graph-v2 — https://github.com/PrefectHQ/prefect/pull/11113

> - Exclude any flow/task run with a `NULL` `start_time` from Graph v2 — https://github.com/PrefectHQ/prefect/pull/11132

Reviewer: These can be referenced in the highlight instead

- Allow tasks to use `get_run_logger` w/o parent flow run — https://github.com/PrefectHQ/prefect/pull/11129
- Allow `ResultFactory` creation `from_task` when no `flow_run_context` — https://github.com/PrefectHQ/prefect/pull/11134

> - Use an unstarted flow run's expected start time for the Graph start time — https://github.com/PrefectHQ/prefect/pull/11138

Reviewer: maybe "in the flow run graph in the Prefect UI"

> - Use an unstarted flow run's expected start time for the Graph start time — https://github.com/PrefectHQ/prefect/pull/11138

Reviewer: This one can be added to the highlight section as well

### Fixes

> - Dont print worker start command for managed pool — https://github.com/PrefectHQ/prefect/pull/11122

Reviewer: ```suggestion
- Avoid printing references to workers when deploying to managed pools — https://github.com/PrefectHQ/prefect/pull/11122
```

### Documentation

> - Fix typo in RELEASE-NOTES.md — https://github.com/PrefectHQ/prefect/pull/11094

Reviewer: ```suggestion
- Fix typo in release notes — https://github.com/PrefectHQ/prefect/pull/11094
```

- Fix docstring for `flow.deploy` method example — https://github.com/PrefectHQ/prefect/pull/11108

> - Add warning to push pool guide — https://github.com/PrefectHQ/prefect/pull/11118

Reviewer: ```suggestion
- Add warning about image architecture to push pool guide — https://github.com/PrefectHQ/prefect/pull/11118
```

> - Move `Webhooks` to `Development` — https://github.com/PrefectHQ/prefect/pull/11141

Reviewer: ```suggestion
- Move webhooks guide to `Development` section in guides index — https://github.com/PrefectHQ/prefect/pull/11141
```

> 
> 

Reviewer: ```suggestion

```
one less blank line
